accounts/abi/bind/dep_tree.go

```diff
@@ -0,0 +1,231 @@
+package bind
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"regexp"
+	"strings"
+)
+
+// DeploymentParams represents parameters needed to deploy a
+// set of contracts.  It takes an optional override
+// list to specify contracts/libraries that have already been deployed on-chain.
+type DeploymentParams struct {
+	Contracts []*MetaData
+	// map of solidity library pattern to constructor input.
+	Inputs map[string][]byte
+	// Overrides is an optional map of pattern to deployment address.
+	// Contracts/libraries that refer to dependencies in the override
+	// set are linked to the provided address (an already-deployed contract).
+	Overrides map[string]common.Address
+}
```

Comment:
This is a public struct, with public fields. It is currently not used/instantiated other than from tests. How do you expect users to use this? 

The tests do it like this, for example: 
```golang
	deploymentParams := bind.DeploymentParams{
		Contracts: append(nested_libraries.C1LibraryDeps, nested_libraries.C1MetaData),
		Inputs:    map[string][]byte{nested_libraries.C1MetaData.Pattern: constructorInput},
		Overrides: nil,
	}
```
It would probably be nicer to do 
```golang
	contracts := append(nested_libraries.C1LibraryDeps, nested_libraries.C1MetaData)
	inputs :=  map[string][]byte{nested_libraries.C1MetaData.Pattern: constructorInput}
	deploymentParams := bind.NewDeploymentParams(contracts, inputs, nil)
```
Does `DeploymentParams` need to be public? Do the members need to be public? Hard to say without knowing how you see it being used by users.

Reply:
> Does DeploymentParams need to be public? Do the members need to be public? Hard to say without knowing how you see it being used by users.

I would say that it should be a public struct.  It's used by `LinkAndDeploy`, so not having it as a struct would just make the method signature more convoluted.

It could have private members, and be instantiated with a constructor as you've suggested.  I don't really feel strongly about this, but my gut says that struct members shouldn't be public without a reason.

Reply:
Yes please make them private. Indeed, struct members should not be public if it can be avoided